src/math/sincos.hpp

```diff
@@ -29,4 +31,24 @@
   #define SINCOS(a,sa,ca) (*sa) = std::sin(a); (*ca) = std::cos(a)
 #endif
 
+/// Compute quaternion and angle from a SO(3) joint configuration
+///
+/// \param q1, q2, robot configurations
+/// \param index index of joint configuration in robot configuration vector
+/// \param unit quaternion corresponding to both joint configuration
+/// \return angle between both joint configuration
+template <typename D>
+static double angleBetweenQuaternions(const Eigen::MatrixBase<D> & q1,
```

Comment:
why static.
You should return D::Scalar and use it as scalar type all along the function